ister.py

```diff
@@ -1387,7 +1387,32 @@ def validate_template(template):
     LOG.debug(template)
 
 
-def process_kernel_cmdline(f_kcmdline, sleep_time=15):
+def download_ister_conf(uri, timeout=15):
+    """Download the ister.conf/ister.json file from 'uri' to a local temporary
+    file and return the temporary file path. The timeout argument specifies for
+    how long to try downloading the file."""
+
+    tmpfd, abs_path = tempfile.mkstemp()
+    LOG.debug("ister_conf tmp file = {0}".format(abs_path))
+
+    start_time = time.time()
+    while True:
+        try:
+            with request.urlopen(uri) as response:
+                with closing(os.fdopen(tmpfd, "wb")) as out_file:
+                    shutil.copyfileobj(response, out_file)
+                    return abs_path
+            os.unlink(abs_path)
+        except Exception as err:
+            # In a PXE environment it's possible systemd launched us before the
+            # network is up. Therefore, keep trying for 'timeout' seconds.
+            if time.time() - start_time < timeout:
+                pass
```

Comment:
Well, this is the same thing - we are in the exept block and 'pass' will just take us out of the exception block and we'll start the new loop iteration. May be 'continue' is a bit more readable, but in my eye there is no difference. I can put 'continue' there.

Reply:
Hrm from the documentation and testing I just went through I'm not sure about pass working the way you describe. Pass is documented as a noop and so it would only impact the if statement because python wants an indentation. The raise Exception following the if statement would still get run.

Maybe my little toy example below isn't quite correct though and I'd be interested in learning what my misunderstanding is (maybe I just shouldn't review on Saturday morning X( ).

`def foo():
    try:
        i = 0
        raise Exception("foo")
    except Exception as err:
        if i == 0:
            pass
        raise Exception("bar")`

And if I run this it raises an exception "bar".

Reply:
**blush**
You are correct. Will fix, thanks.